imap_processing/codice/codice_l1a.py

```diff
@@ -372,65 +406,73 @@ def process_codice_l1a(file_path: Path | str) -> xr.Dataset:
-            pipeline.get_lo_data_products(apid)
-            pipeline.unpack_science_data(science_values)
-            dataset = pipeline.create_science_dataset(start_time)
-
-        elif apid == CODICEAPID.COD_LO_PHA:
-            logger.info(f"{apid} is currently not supported")
-            continue
-
-        elif apid == CODICEAPID.COD_HI_INST_COUNTS_AGGREGATED:
-            logger.info(f"{apid} is currently not supported")
-            continue
-
-        elif apid == CODICEAPID.COD_HI_INST_COUNTS_SINGLES:
-            logger.info(f"{apid} is currently not supported")
-            continue
-
-        elif apid == CODICEAPID.COD_HI_PHA:
-            logger.info(f"{apid} is currently not supported")
-            continue
-
-        elif apid == CODICEAPID.COD_HI_OMNI_SPECIES_COUNTS:
-            logger.info(f"{apid} is currently not supported")
-            continue
-
-        elif apid == CODICEAPID.COD_HI_SECT_SPECIES_COUNTS:
-            logger.info(f"{apid} is currently not supported")
-            continue
+    # Temporary workaround in order to create hi data products in absence of
+    # simulated data
+    if file_path.name.startswith("imap_codice_l0_lo") or file_path.name.startswith(
+        "imap_codice_l0_hskp"
+    ):
+        # Decom the packets, group data by APID, and sort by time
+        packets = decom_packets(file_path)
+        grouped_data = group_by_apid(packets)
+
+        for apid in grouped_data.keys():
+            logger.info(f"\nProcessing {CODICEAPID(apid).name} packet")
+
+            if apid == CODICEAPID.COD_NHK:
+                packets = grouped_data[apid]
+                sorted_packets = sort_by_time(packets, "SHCOARSE")
+                dataset = create_hskp_dataset(packets=sorted_packets)
+
+            elif apid in apids_for_lo_science_processing:
+                # Sort the packets by time
+                packets = sort_by_time(grouped_data[apid], "SHCOARSE")
+
+                # Determine the start time of the packet
+                start_time = calc_start_time(
+                    packets[0].data["ACQ_START_SECONDS"].raw_value,
+                    launch_time=np.datetime64("2010-01-01T00:01:06.184", "ns"),
+                )
+
+                # Extract the data
+                science_values = packets[0].data["DATA"].raw_value
+
+                # Get the four "main" parameters for processing
+                table_id, plan_id, plan_step, view_id = get_params(packets[0])
+
+                # Run the pipeline to create a dataset for the product
+                pipeline = CoDICEL1aPipeline(table_id, plan_id, plan_step, view_id)
+                pipeline.get_esa_sweep_values()
+                pipeline.get_acquisition_times()
+                pipeline.get_data_products(apid)
+                pipeline.unpack_science_data(science_values)
+                dataset = pipeline.create_science_dataset(start_time)
+
+    # Temporary workaround in order to create hi data products in absence of
```

Comment:
Add some kind of tag so we can easily come back to this later and know where to look for updates that are needed.
```suggestion
    # TODO: Temporary workaround in order to create hi data products in absence of
```